test/integration/rules/rules_test.rb

```diff
@@ -0,0 +1,57 @@
+require 'test_helper'
+Dir.glob(Rails.root + 'test/fixtures/rule_fixtures/*.rb').each do |fixture|
+	require_relative fixture
+end
+
+class MagiRulesTest < ActionDispatch::IntegrationTest
+	include ApplicationProcessor
+	include ApplicationParser
+
+	# dynamically load and transform names of everything in fixtures/rule_fixtures
+	@all_fixtures = Dir.entries(Rails.root + 'test/fixtures/rule_fixtures').select { |file| file.to_s.length > 2 }.map do |file|
+		file.gsub('.rb','').split('_').map { |x| /chip/i.match(x) ? x.upcase : x.capitalize }.join('')
+	end
```

Comment:
Hmm, I think there's a smoother way to do this. Might be worth looking at how all the rules get loaded in the application.